python_modules/dagster/dagster/_core/definitions/op_definition.py

```diff
@@ -481,35 +481,41 @@ def _resolve_output_defs_from_outs(
         only_out = outs[name]
         return [only_out.to_definition(annotation, name, description, default_code_version)]
 
-    output_defs: List[OutputDefinition] = []
+    # If no type hint or a result object type, use empty annotation
```

Comment:
does this need to handle `AssetCheckResult`?  What about a future `ObservationResult`?

Reply:
> does this need to handle AssetCheckResult?

I dont think so, singular `AssetCheckResult` should only be from `@asset_check` . i could test it 

> What about a future ObservationResult?

yes i assume so

Reply:
can we make this a helper function or do something to make it clear where a future result type should live?